lib/string_reverse.rb

```diff
@@ -1,4 +1,22 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+
+  if my_string == nil
+    return nil
+  elsif my_string.length == 1
+    return my_string
+  elsif
+    my_string.length > 1
+    temp = ""

```

Comment:
Pedantic: Technically, you don't need temp outside the loop at all. So you don't need to define it here.